pkg/kubernetes/client.go

```diff
@@ -360,3 +409,66 @@ func (r *RawResource) CreatePatchOrReplace() error {
 	msg("patched")
 	return nil
 }
+
+// Delete attempts to delete this Kubernetes resource, or returns doing nothing
+// if it does not exist.
+func (r *RawResource) Delete() (string, error) {
+	exists, _, err := r.exists()
+	if err != nil {
+		return "", err
+	}
+	if !exists {
+		return "", nil
+	}
+	propagationPolicy := metav1.DeletePropagationForeground
+	if _, err := r.Helper.DeleteWithOptions(r.Info.Namespace, r.Info.Name, &metav1.DeleteOptions{
+		PropagationPolicy: &propagationPolicy,
+	}); err != nil {
+		return "", err
+	}
+	if err := r.waitForDeletion(); err != nil {
```

Comment:
Shouldn't we make synchronous operations optional? Otherwise it would be more readable to call the function `DeleteSynchronously()`

Reply:
I also considered that, but since we don't need it now... 🙂 
I'll rename then. We can refine when we actually need a `Delete` which is asynchronous.

Reply:
Yes, it does look like we should make it optional or rename the function. If you chose to rename, please use `DeleteSync`, as that follows the naming used in `pkg/cfn/manager` (e.g. `tasks.DoAllSync` and `DeleteStackBySpecSync`).

Reply:
Damn GitHub, your reply didn't show up @errordeveloper, renaming again.